edith-step/src/App.js

```diff
@@ -1,19 +1,81 @@
 import React, {Component} from 'react';
-
+import LineChart, {BarGraph, CategoryDoughnutChart,
+  ItemDoughnutChart} from './UserChart';
 import ReceiptInput from './components/ReceiptInput';
 import TopNavbar from './components/TopNavbar';
-
 import './App.css';
 
-/**
- * Main webpage for the website.
- */
+/** Corresponds to the different chart types. */
+const chart = {
+  'LINE': 'LINE',
+  'BAR': 'BAR',
+  'DOUGHNUT': 'DOUGHNUT',
+};
+
+/** Main webpage for the website. */
 class App extends Component {
+  /** Constructor */
+  constructor() {
+    super();
+    this.state = {'chartType': LineChart, 'dateFilter': '',
+      'categoryFilter': ''};
+    this.updateChartType = this.updateChartType.bind(this);
+    this.showItemChart = this.showItemChart.bind(this);
```

Comment:
previous comment of using arrow function for  ```showItemChart``` and ```revertCharts```

Reply:
Done.